packages/hydrogen/src/seo/generate-seo-tags.ts

```diff
@@ -335,225 +356,235 @@ export function generateSeoTags<
+            }),
+          );
+        }
 
+        break;
+      }
+
+      case 'robots': {
+        if (!seoInput.robots) {
           break;
         }
 
-        case 'robots':
-          // Robots
-          const {
-            maxImagePreview,
-            maxSnippet,
-            maxVideoPreview,
-            noArchive,
-            noFollow,
-            noImageIndex,
-            noIndex,
-            noSnippet,
-            noTranslate,
-            unavailableAfter,
-          } = value as RobotsOptions;
-
-          const robotsParams = [
-            noArchive && 'noarchive',
-            noImageIndex && 'noimageindex',
-            noSnippet && 'nosnippet',
-            noTranslate && `notranslate`,
-            maxImagePreview && `max-image-preview:${maxImagePreview}`,
-            maxSnippet && `max-snippet:${maxSnippet}`,
-            maxVideoPreview && `max-video-preview:${maxVideoPreview}`,
-            unavailableAfter && `unavailable_after:${unavailableAfter}`,
-          ];
-
-          let robotsParam =
-            (noIndex ? 'noindex' : 'index') +
-            ',' +
-            (noFollow ? 'nofollow' : 'follow');
-
-          for (let param of robotsParams) {
-            if (param) {
-              robotsParam += `,${param}`;
-            }
+        const {
+          maxImagePreview,
+          maxSnippet,
+          maxVideoPreview,
+          noArchive,
+          noFollow,
+          noImageIndex,
+          noIndex,
+          noSnippet,
+          noTranslate,
+          unavailableAfter,
+        } = seoInput.robots;
+
+        const robotsParams = [
+          noArchive && 'noarchive',
+          noImageIndex && 'noimageindex',
+          noSnippet && 'nosnippet',
+          noTranslate && `notranslate`,
+          maxImagePreview && `max-image-preview:${maxImagePreview}`,
+          maxSnippet && `max-snippet:${maxSnippet}`,
+          maxVideoPreview && `max-video-preview:${maxVideoPreview}`,
+          unavailableAfter && `unavailable_after:${unavailableAfter}`,
+        ];
+
+        let robotsParam =
+          (noIndex ? 'noindex' : 'index') +
+          ',' +
+          (noFollow ? 'nofollow' : 'follow');
+
+        for (let param of robotsParams) {
+          if (param) {
+            robotsParam += `,${param}`;
           }
+        }
 
-          tagResults.push(
-            generateTag('meta', {name: 'robots', content: robotsParam}),
-          );
+        tagResults.push(
+          generateTag('meta', {name: 'robots', content: robotsParam}),
+        );
 
-          break;
+        break;
       }
+      case 'jsonLd': {
+        const content = seoInput.jsonLd;
 
-      return tagResults;
-    });
+        if (typeof content !== 'object') {
+          break;
+        }
 
-    const entries = tags.flat();
+        jsonLd = {...jsonLd, ...content};
 
-    output.push(
-      // @ts-expect-error untyped
-      entries.filter((value) => !!value),
-    );
+        break;
+      }
```

Comment:
I think we can add a type check here for never to make sure we've exhausted all options